goutubedl_test.go

```diff
@@ -301,3 +301,122 @@ func TestOptionDownloader(t *testing.T) {
 	}
 	dr.Close()
 }
+
+func TestInvalidOptionTypeField(t *testing.T) {
+	defer leakChecks(t)()
+
+	_, err := goutubedl.New(context.Background(), playlistRawURL, goutubedl.Options{
+		Type: 42,
+	})
+	if err == nil {
+		t.Error("should have failed")
+	}
+}
+
+func TestDownloadPlaylistEntry(t *testing.T) {
+	defer leakChecks(t)()
+	// Download file by specifying the playlist index
+	stderrBuf := &bytes.Buffer{}
+	r, err := goutubedl.New(context.Background(), playlistRawURL, goutubedl.Options{
+		StderrFn: func(cmd *exec.Cmd) io.Writer {
+			return stderrBuf
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedTitle := "Kindred Phenomena"
+	if r.Info.Title != expectedTitle {
+		t.Errorf("expected title %q got %q", expectedTitle, r.Info.Title)
+	}
+
+	expectedEntries := 8
+	if len(r.Info.Entries) != expectedEntries {
+		t.Errorf("expected %d entries got %d", expectedEntries, len(r.Info.Entries))
+	}
+
+	expectedTitleOne := "B1 Mattheis - Ben M"
+	playlistIndex := 2
+	if r.Info.Entries[playlistIndex].Title != expectedTitleOne {
+		t.Errorf("expected title %q got %q", expectedTitleOne, r.Info.Entries[playlistIndex].Title)
+	}
+
+	dr, err := r.DownloadWithOptions(context.Background(), goutubedl.DownloadOptions{
+		PlaylistIndex: playlistIndex + 1,
```

Comment:
Could this use `PlaylistIndex` so something like `r.Info.Entries[playlistIndex].PlaylistIndex`? that should work or? but annoying PlaylistIndex is a float64 hmm